File: result/metagpt/claude/humaneval/56.c
#include <stdbool.h>
#include <string.h>

bool correct_bracketing(const char* brackets) {
    int level = 0;
    int length = strlen(brackets);
    for (int i = 0; i < length; i++) {
        if (brackets[i] == '<') level += 1;
        if (brackets[i] == '>') level -= 1;
        if (level < 0) return false;
    }
    if (level != 0) return false;
    return true;
}

File: result/metagpt/deepseek/humaneval/56.rs
pub fn correct_bracketing(brackets: &str) -> bool {
    let mut level = 0;
    for c in brackets.chars() {
        match c {
            '<' => level += 1,
            '>' => {
                level -= 1;
                if level < 0 {
                    return false;
                }
            },
            _ => ()
        }
    }
    level == 0
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_correct_bracketing() {
        assert!(correct_bracketing("<<>>"));
        assert!(!correct_bracketing("<><>"));
        assert!(!correct_bracketing("<<>"));
    }
}